ahnlich/client/src/ai.rs

```diff
@@ -0,0 +1,665 @@
+            AIStoreType::RawString,
+            AIModel::Llama3,
+            HashSet::new(),
+            HashSet::new(),
+        );
+        pipeline.create_store(
+            StoreName("Main2".to_string()),
+            AIStoreType::RawString,
+            AIModel::Llama3,
+            HashSet::new(),
+            HashSet::new(),
+        );
+        pipeline.create_store(
+            StoreName("Less".to_string()),
+            AIStoreType::Binary,
+            AIModel::Llama3,
+            HashSet::new(),
+            HashSet::new(),
+        );
+        pipeline.list_stores();
+        pipeline.drop_store(StoreName("Less".to_string()), true);
+        pipeline.purge_stores();
+        let mut expected = AIServerResult::with_capacity(6);
+        expected.push(Ok(AIServerResponse::Unit));
+        expected.push(Ok(AIServerResponse::Unit));
+        expected.push(Ok(AIServerResponse::Unit));
+        expected.push(Ok(AIServerResponse::StoreList(HashSet::from_iter([
+            AIStoreInfo {
+                name: StoreName("Main".to_string()),
+                embedding_size: AIModel::Llama3.embedding_size().into(),
+                r#type: AIStoreType::RawString,
+                model: AIModel::Llama3,
+            },
+            AIStoreInfo {
+                name: StoreName("Main2".to_string()),
+                embedding_size: AIModel::Llama3.embedding_size().into(),
+                r#type: AIStoreType::RawString,
+                model: AIModel::Llama3,
+            },
+            AIStoreInfo {
+                name: StoreName("Less".to_string()),
+                embedding_size: AIModel::Llama3.embedding_size().into(),
+                r#type: AIStoreType::Binary,
+                model: AIModel::Llama3,
+            },
+        ]))));
+        expected.push(Ok(AIServerResponse::Del(1)));
+        expected.push(Ok(AIServerResponse::Del(2)));
+        let res = pipeline.exec().await.expect("Could not execute pipeline");
+        assert_eq!(res, expected);
+    }
+
+    #[tokio::test]
+    async fn test_ai_proxy_get_pred() {
+        let address = provision_test_servers().await;
+
+        let host = address.ip();
+        let port = address.port();
+        let ai_client = AIClient::new(host.to_string(), port)
+            .await
+            .expect("Could not initialize client");
+
+        let store_name = StoreName(String::from("Deven Kicks"));
+        let matching_metadatakey = MetadataKey::new("Brand".to_owned());
+        let matching_metadatavalue = MetadataValue::RawString("Nike".to_owned());
+
+        let nike_store_value =
+            StoreValue::from_iter([(matching_metadatakey.clone(), matching_metadatavalue.clone())]);
+        let adidas_store_value = StoreValue::from_iter([(
+            matching_metadatakey.clone(),
+            MetadataValue::RawString("Adidas".to_owned()),
+        )]);
+        let store_data = vec![
+            (
+                StoreInput::RawString(String::from("Jordan 3")),
```

Comment:
I'm interested in exploring the branch of Binary examples so as much as possible, also make queries with `StoreInput::Binary`